intg/src/main/java/org/apache/atlas/model/lineage/AtlasLineageOnDemandInfo.java

```diff
@@ -0,0 +1,407 @@
+package org.apache.atlas.model.lineage;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import org.apache.atlas.model.instance.AtlasEntityHeader;
+
+import javax.xml.bind.annotation.XmlAccessType;
+import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlRootElement;
+import java.io.Serializable;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+import static com.fasterxml.jackson.annotation.JsonAutoDetect.Visibility.NONE;
+import static com.fasterxml.jackson.annotation.JsonAutoDetect.Visibility.PUBLIC_ONLY;
+
+@JsonAutoDetect(getterVisibility = PUBLIC_ONLY, setterVisibility = PUBLIC_ONLY, fieldVisibility = NONE)
+@JsonSerialize(include = JsonSerialize.Inclusion.NON_NULL)
+@JsonIgnoreProperties(ignoreUnknown = true, value = {"visitedEdges", "skippedEdges"})
+@XmlRootElement
+@XmlAccessorType(XmlAccessType.PROPERTY)
+public class AtlasLineageOnDemandInfo implements Serializable {
+    private String baseEntityGuid;
+    private LineageDirection lineageDirection;
+    private int lineageDepth;
+    private Map<String, AtlasEntityHeader> guidEntityMap;
+    private Set<LineageRelation> relations;
+    private Set<String>                             visitedEdges;
+    private Set<String>                             skippedEdges;
+    private Map<String, LineageInfoOnDemand>        relationsOnDemand;
+    private Map<String, LineageOnDemandConstraints> lineageOnDemandPayload;
```

Comment:
api response should not just have LineageOnDemandConstraints, it should have other items as well from the payload

Reply:
addressed